pkg/source/gcp/task/cloud-composer/form/autocomplete.go

```diff
@@ -29,44 +27,69 @@ import (
 	"github.com/GoogleCloudPlatform/khi/pkg/task/taskid"
 )
 
+// This is an implementation for gcp_task.AutocompleteClusterNamesTaskID
+// the task returns GKE cluster name where the provided Composer environment is running
 var AutocompleteClusterNames = inspection_cached_task.NewCachedTask(composer_taskid.AutocompleteClusterNamesTaskID, []taskid.UntypedTaskReference{
 	gcp_task.InputProjectIdTaskID,
+	composer_taskid.InputComposerEnvironmentTaskID,
 }, func(ctx context.Context, prevValue inspection_cached_task.PreviousTaskResult[*gcp_task.AutocompleteClusterNameList]) (inspection_cached_task.PreviousTaskResult[*gcp_task.AutocompleteClusterNameList], error) {
+
 	client, err := api.DefaultGCPClientFactory.NewClient()
 	if err != nil {
 		return inspection_cached_task.PreviousTaskResult[*gcp_task.AutocompleteClusterNameList]{}, err
 	}
 
 	projectID := task.GetTaskResult(ctx, gcp_task.InputProjectIdTaskID.GetTaskReference())
-	if projectID != "" && projectID == prevValue.DependencyDigest {
+	environment := task.GetTaskResult(ctx, composer_taskid.InputComposerEnvironmentTaskID.GetTaskReference())
+	dependencyDigest := environment
```

Comment:
The `dependencyDigest` should be concatenate of `environment` and `projectID`.
Because the cache must be refreshed when users modified project ID.

Reply:
you right, I missed it. Let me fix it.